### PR TITLE
Add lint codes to a few more lints

### PR DESCRIPTION
**Pull request by bwilkerson**
After this, there are only 7 lints remaining that don't have a `LintCode` defined.

### CODE REVIEW

File: lib/src/rules/diagnostic_describe_all_properties.dart [hunk 1]
```diff
@@ -62,11 +62,17 @@ class Absorber extends Widget {
...
+}
 ```
 ''';
 
 class DiagnosticsDescribeAllProperties extends LintRule {
+  static const LintCode code = LintCode(
+      'diagnostic_describe_all_properties',
+      "The public property isn't described by either 'debugFillProperties' or "
+          "'debugDescribeChildren'.",
+      correctionMessage: 'Try described the property.');
```
Comment (srawlins) [inline lib/src/rules/diagnostic_describe_all_properties.dart:74]: "describing"?

File: lib/src/rules/discarded_futures.dart [hunk 2]
```diff
@@ -44,13 +44,22 @@ Future<void> createDir(String path) async {}
 ''';
 
 class DiscardedFutures extends LintRule {
+  static const LintCode code = LintCode('discarded_futures',
+      "Asynchronous function invoked in a non-'async' function.",
+      correctionMessage:
+          "Try converting the enclosing function to be 'async' and then "
+          "'await' the future");
```
Comment (srawlins) [inline lib/src/rules/discarded_futures.dart:51]: Should Future be capitalized?

**Review by srawlins — APPROVED** [review]

File: lib/src/rules/prefer_final_locals.dart [hunk 3]
```diff
@@ -46,6 +46,10 @@ void mutableCase() {
 ''';
 
 class PreferFinalLocals extends LintRule {
+  static const LintCode code = LintCode(
+      'prefer_final_locals', 'Local variables should be final.',
+      correctionMessage: 'Try making the variables final.');
```
Comment (pq) [inline lib/src/rules/prefer_final_locals.dart:51]: I think this is reported on each variable, in which case maybe 

    'Try making the variable final.'

**Review by pq — APPROVED** [review]

File: lib/src/rules/diagnostic_describe_all_properties.dart (hunk 1 above)
Comment (bwilkerson) [reply lib/src/rules/diagnostic_describe_all_properties.dart]: Done.

File: lib/src/rules/discarded_futures.dart (hunk 2 above)
Comment (bwilkerson) [reply lib/src/rules/discarded_futures.dart:51]: In other docs I've written (not sure whether there's a standard), I've capitalized "Future" when referring to the class, but not when referring to the concept. This is a reference to the concept, hence the lowercase letter.

File: lib/src/rules/prefer_final_locals.dart (hunk 3 above)
Comment (bwilkerson) [reply lib/src/rules/prefer_final_locals.dart]: Done.

**Pull request merged**